### PR TITLE
#83 - Display Page Proposed Solutions Visual

### LINKED ISSUE #83: Change Requests - Display Page Proposed Solutions Visual
## Description

Add to the regular change request page necessary components to display existing proposed solutions to a change request.

## Prerequisite

- [x] Northeastern-Electric-Racing/FinishLine#84 

## Business Logic

On the Change Request details page, add a new section for the proposed solutions. This form should display existing solutions, as well as add the ability to add new proposed solutions. The details of this block should follow from ticket Northeastern-Electric-Racing/FinishLine#82 

This section should be a box below the details box, but above the Review Notes section. At the bottom / top of this box should be the Add New Proposed Solution button. Each proposed solution should be the read only version of the proposed solution form. The box should be scrollable.

At the top of the box, there should always be a button to add a new proposed solution, with the label "Add Proposed Solution"
  - [x] This button should have a "+" symbol to the left of the text
- [x] When clicking the button, open a modal to enter the information
  - [x] This modal should consist of the existing individual proposed solution form, in the editing mode
  - [x] Below this, there should be an "Add" button that adds the data into the list of proposed solutions
- [x] Above this, there should be a list of edit proposed solution forms, filling up 60-80 percent of the page. They should be in read only form
- [x] These boxes should go from oldest to newest, with the newest at the top, and the oldest at the bottom

## Mockup

Change_Request_Details_Page

### PR DESCRIPTION
**Pull request by Ari-S-123**
## Changes

Standard CRs now have a list of proposed solutions displayed on the details page.

## Screenshots (if applicable)

Screenshot 2022-09-19 140824

Screenshot 2022-09-09 000446

Screenshot 2022-09-19 140734

Screenshot 2022-09-19 140624

Screenshot 2022-09-19 135719

## To Do

_Any remaining things that need to get done_

- [x] Unit Testing

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://github.com/Northeastern-Electric-Racing/FinishLine/blob/develop/docs/ContributorGuide.md) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors
- [x] No newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (if applicable)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] PR is linked to the ticket
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack

Closes #83

### CODE REVIEW

**Comment by jamescd18** [comment]
I think the third image looks really gross for a UI. I think it looks worse to be displaying proposed solutions via read only form inputs. I also don't think it makes any sense because the form inputs will never not be read only

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
yeah I agree with James' review that they shouldn't be read only inputs. Just display them normally as text. I think this will require making a new component (like a ProposedSolutionView and changing the other one to ProposedSolutionForm or something).

I also agree with James that this is probs where we should remove the budget/timeline/scope aspect for CRs that now don't matter

**Review by RChandler234 — DISMISSED** [review]

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx [hunk 1]
```diff
@@ -46,111 +40,109 @@ const ProposedSolutionForm: React.FC<ProposedSolutionFormProps> = ({
...
-          </Row>
-        </Form>
-      </PageBlock>
-    </>
+    <PageBlock title="" cardContainerStyle="mb-0">
+      <Form id="individual-proposed-solution-form" onSubmit={handleSubmit(onAdd)}>
+        <Row className="mx-2 justify-content-start">
+          <Col lg={true}>
+            <Form.Group controlId="formDescription" className="mx-2">
+              <Form.Label>Description</Form.Label>
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx:58]: these labels should show up as white in dark mode

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
In addition to my other comment two more things:
1. the add new proposed solution button shouldn't show up if the cr has already been reviewed (approved or denied)
2. unused import in ChangeRequestDetailsView.test.tsx

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        </Button>
+      ) : (
+        ''
+      )}
+      <div
+        style={{
+          maxHeight: '35em',
+          overflowY: 'auto',
+          display: 'flex',
+          flexDirection: 'column-reverse'
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx:43]: This is too much to have inline like this, I'd either make a style object than just call it here, or move this stuff into  a css file

**Review by anthonybernardi — COMMENTED** [review]
the pill shouldn't be on a separate row, that's how it looks rn. ALso I think we should copy Kevin's layout for each proposed solution

**Comment by anthonybernardi** [comment]
Screen Shot 2022-09-19 at 3 24 58 PM
screenshot of changes i made to make it spaced better

**Pull request merged**